### DISCUSSION TITLE
Vtm render-theme with pg_tileserv

### DISCUSSION THREAD

**Discussion by Neo003** [mapsforge/vtm#932] [Q&A] [answered]
Hi everyone,

I try to use VTM on Android in order to display OSM vector tiles from a pg_tileserv server.

I've loaded some OSM data from Geofabrik into a PostGIS database with `osm2pgsql` tool with a command like: `osm2pgsql -U postgres -W -d osmDatabase -H localhost france-latest.osm.pbf`. This command generates `planet_osm_line, planet_osm_point, planet_osm_polygon` and `planet_osm_roads` tables without errors.

I've checked that pg_tileserv serves tile layers correctly by  using the web interface at `http://localhost:7800`. I can see tiles without styling (everything is rendered in blue color, but it's correct). 

Next, in an Android Studio project based on the MapilionMvtActivity, I've used a modified MapilionMvtTileSource (I've changed the DEFAULT_URL to point my pg_tileserv url).

Finally, I've studied the render-theme documentation and the `default.xml` file rules but I don't understand why my map never shows the vector tiles from my pg_tileserv server with the Vtm default theme. Actually, none of the embedded themes work.
I think I've misunderstood something about the render-theme rules but I don't find what.

Can someone help me ?

Thanks.

**Comment by devemux86** [comment]
The render themes render the map elements based on rules that use tags,
the official OpenStreetMap tags or modified (as OpenMapTiles / Mapilion).

**Reply by Neo003** [reply]
Thanks for your response. As a base for studying, can you show me an example of a render theme code with official OSM tags ?
Thanks

**Comment by devemux86** [comment, marked as answer]
> As a base for studying, can you show me an example of a render theme code with official OSM tags ?

- The [Default](https://github.com/mapsforge/vtm/blob/master/vtm-themes/resources/assets/vtm/default.xml) or [Osmarender](https://github.com/mapsforge/vtm/blob/master/vtm-themes/resources/assets/vtm/osmarender.xml) themes use the official OpenStreetMap tags.
[Mapsforge maps](https://download.mapsforge.org/) are created with the official OpenStreetMap tags ([tag-mapping](https://github.com/mapsforge/mapsforge/blob/master/mapsforge-map-writer/src/main/config/tag-mapping.xml)).
- There is the [OpenMapTiles](https://github.com/mapsforge/vtm/blob/master/vtm-themes/resources/assets/vtm/openmaptiles.xml) theme used in Mapilion, because of the [OpenMapTiles vector tile schema](https://openmaptiles.org/schema/) changes.
